### PR TITLE
[WIP] Dockerfile: Use US mirror for Debian to make PowerPC and Z more stable

### LINKED ISSUE #38460: CI: PowerPC / Z Could not resolve cdn-fastly.deb.debian.org
CI build on PowerPC looks to be failing to this error time to time:
```
09:38:01 Err:1 http://cdn-fastly.deb.debian.org/debian stretch InRelease
09:38:01   Could not resolve 'cdn-fastly.deb.debian.org'
09:38:01 Err:2 http://security.debian.org/debian-security stretch/updates InRelease
09:38:01   Could not resolve 'security.debian.org'
09:38:01 Err:3 http://cdn-fastly.deb.debian.org/debian stretch-updates InRelease
09:38:01   Could not resolve 'cdn-fastly.deb.debian.org'
09:38:01 Reading package lists...
09:38:01 W: Failed to fetch http://cdn-fastly.deb.debian.org/debian/dists/stretch/InRelease  Could not resolve 'cdn-fastly.deb.debian.org'
09:38:01 W: Failed to fetch http://security.debian.org/debian-security/dists/stretch/updates/InRelease  Could not resolve 'security.debian.org'
09:38:01 W: Failed to fetch http://cdn-fastly.deb.debian.org/debian/dists/stretch-updates/InRelease  Could not resolve 'cdn-fastly.deb.debian.org'
09:38:01 W: Some index files failed to download. They have been ignored, or old ones used instead.
```

Example on https://jenkins.dockerproject.org/job/Docker-PRs-powerpc/12743/console

### PR DESCRIPTION
**Pull request by olljanat**
**- What I did**
Switched Dockerfile to use US mirror for Debian to make PowerPC and Z more stable.

Closes #38460

### CODE REVIEW

**Comment by cpuguy83** [comment]
Shouldn't deb.debian.org should be finding the best mirror already?

Ping @tianon

**Comment by olljanat** [comment]
Can be. Especially if some one can investigate why PowerPC and Z time to time fails to resolve default mirror from DNS.

Will try re-run this when there is those failures on other PRs.

**Comment by tianon** [comment]
http://deb.debian.org should be a combination of fastly and cloudfront, so yes, it should be pretty fast (and Geo-redundant) already, and definitely more reliable than any single mirror (or even mirror round robin) is typically going to be.

If the issue is DNS-related, then switching mirrors probably won't help (since DNS issues will likely still persist and don't typically have much to do with the server you're trying to do a lookup for).

File: Dockerfile [hunk 1]
```diff
@@ -26,8 +26,8 @@
 
 FROM golang:1.11.4 AS base
 # allow replacing httpredir or deb mirror
-ARG APT_MIRROR=deb.debian.org
-RUN sed -ri "s/(httpredir|deb).debian.org/$APT_MIRROR/g" /etc/apt/sources.list
+ARG APT_MIRROR2=ftp.us.debian.org
```
Comment (AkihiroSuda) [inline Dockerfile]: I think Jenkins should override `APT_MIRROR` with `docker build --build-arg`

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (olljanat) [reply Dockerfile]: Yes it does. That why I needed to change parameter to be able to test this.

**Pull request closed without merging**